patches/api/0347-Add-Shearable-API.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 17 Oct 2021 15:39:41 -0400
+Subject: [PATCH] Add Shearable API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/Shearable.java b/src/main/java/io/papermc/paper/entity/Shearable.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..dd2d00b3dfeca5400a86d8d6b78e1e1d74c71d8d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/Shearable.java
+@@ -0,0 +1,30 @@
++package io.papermc.paper.entity;
++
++import net.kyori.adventure.sound.Sound;
++import org.jetbrains.annotations.NotNull;
++
++public interface Shearable {
++
++    /**
++     * Forces the entity to play the effect of being sheared by a player.
```

Comment:
I feel that its wrong to put "play the effect of being sheared" before it *actually* being sheared. The method is `shear`, the javadoc should first state, this will shear the entity, and add that it plays associated effects